packages/core/core/src/public/Config.js

```diff
@@ -120,7 +120,10 @@ export default class PublicConfig implements IConfig {
     }
 
     if (!options || !options.exclude) {
-      if (this.#config.resolvedPath == null) {
+      if (
+        this.#config.resolvedPath == null ||
```

Comment:
I'm kind of confused as to how this works, how do you know the first request to `getConfig` is the actual config and not a package.json or something as it was before?

cc @padmaia

Reply:
I'm questioning whether we really need to distinguish between `resolvedPath` and `includedFiles`. Currently Parcel core uses `resolvedPath` to know where to resolve from to get the version of the `devDeps`, but if any of the `devDeps` are defined in the included files, the plugin would need to resolve the version itself. Maybe we should just make them do that in all cases and only use `includedFiles`? If we do that, the only use of `devDeps` would be to set up an invalidation when the lock file changes. Maybe we could also get rid of them and make plugins add the lockfile to `includedFiles`? Would make the api a tad simpler. It would also eliminate the need for a bunch of devDep request nodes. Thoughts?

cc @devongovett

Reply:
Sounds better like that, having everything in includedfiles will be simpler and more stable you have the package.json location for resolving devDeps, like you mentiones we could also add the location of the lockfiles or something.

If @devongovett agrees this is a good idea I can update this PR to do exactly that or we can merge this for now and do it in a follow-up

Reply:
I like that idea